extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -450,43 +509,88 @@ public void onValueReceived(String variable, String value, String service) {
+        return oldValue != null ? !oldValue.equals(value) : value != null;
+    }
+
+    /**
+     * Similar to the AVTransport eventing, the Queue events its state variables
+     * as sub values within a synthesized LastChange state variable.
+     */
+    private boolean isQueueEvent(String variable) {
+        return "LastChange".equals(variable);
+    }
+
+    private void updateGroupCoordinator() {
+        updateState(new ChannelUID(getThing().getThingTypeUID(), getThing().getUID(), COORDINATOR),
+                new StringType(getCoordinator()));
+        coordinatorHandler = getHandlerByName(getCoordinator());
+    }
+
+    private void removeSubscription() {
+        synchronized (upnpLock) {
+            // Set up GENA Subscriptions
+            if (service.isRegistered(this)) {
+                for (String subscription : SERVICE_SUBSCRIPTIONS) {
+                    service.removeSubscription(this, subscription);
+                }
+                service.unregisterParticipant(this);
             }
-            service.unregisterParticipant(this);
         }
     }
 
-    private synchronized void onUpdate() {
-        if (service.isRegistered(this)) {
-            if (pollingJob == null || pollingJob.isCancelled()) {
-                Configuration config = getThing().getConfiguration();
-                // use default if not specified
-                int refreshInterval = DEFAULT_REFRESH_INTERVAL;
-                Object refreshConfig = config.get("refresh");
-                if (refreshConfig != null) {
-                    refreshInterval = ((BigDecimal) refreshConfig).intValue();
+    private void onUpdate() {
+        synchronized (upnpLock) {
+            if (service.isRegistered(this)) {
+                for (String subscription : SERVICE_SUBSCRIPTIONS) {
+                    service.addSubscription(this, subscription, SUBSCRIPTION_DURATION);
+                }
+
+                if (pollingJob == null || pollingJob.isCancelled()) {
+                    Configuration config = getThing().getConfiguration();
+                    // use default if not specified
+                    int refreshInterval = DEFAULT_REFRESH_INTERVAL;
+                    Object refreshConfig = config.get("refresh");
+                    if (refreshConfig != null) {
+                        refreshInterval = ((BigDecimal) refreshConfig).intValue();
+                    }
+                    pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0, refreshInterval, TimeUnit.SECONDS);
                 }
-                pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0, refreshInterval, TimeUnit.SECONDS);
             }
         }
     }
 
+    private void updatePlayerState() {
+        Map<String, String> result = service.invokeAction(this, "DeviceProperties", "GetZoneInfo", null);
+        if (result.isEmpty()) {
+            logger.debug("Sonos player " + getThing().getProperties().get(Thing.PROPERTY_MODEL_ID)
```

Comment:
See the error output here attached. I get a lot of these, and cfr this line, the Thing.PROPERTY_MODEL_ID seems to be empty with all of them

Apr 11, 2016 5:02:00 PM org.jupnp.transport.spi.AbstractStreamClient sendRequest
INFO: Timeout of 10 seconds while waiting for HTTP request to complete, aborting: (OutgoingSubscribeRequestMessage) SUBSCRIBE http://192.168.0.161:1400/AudioIn/Event
Apr 11, 2016 5:02:00 PM org.jupnp.transport.spi.AbstractStreamClient sendRequest
INFO: Timeout of 10 seconds while waiting for HTTP request to complete, aborting: (OutgoingActionRequestMessage) POST http://192.168.0.153:1400/MediaRenderer/AVTransport/Control
2016-04-11 17:02:00.203 [DEBUG] [e.s.i.t.upnp.UpnpIOServiceImpl:276  ] - Current state of service prevents invoking that action. Connection error or no response received.
Apr 11, 2016 5:02:00 PM org.jupnp.transport.spi.AbstractStreamClient sendRequest
INFO: Timeout of 10 seconds while waiting for HTTP request to complete, aborting: (OutgoingActionRequestMessage) POST http://192.168.0.156:1400/MediaRenderer/AVTransport/Control
2016-04-11 17:02:00.225 [DEBUG] [e.s.i.t.upnp.UpnpIOServiceImpl:276  ] - Current state of service prevents invoking that action. Connection error or no response received.
2016-04-11 17:02:00.226 [DEBUG] [e.s.i.t.upnp.UpnpIOServiceImpl:270  ] - Invoking Action 'GetZoneInfo' of service 'DeviceProperties' for participant 'RINCON_000E58D6F1A601400'
Apr 11, 2016 5:02:00 PM org.jupnp.transport.spi.AbstractStreamClient sendRequest
INFO: Timeout of 10 seconds while waiting for HTTP request to complete, aborting: (OutgoingActionRequestMessage) POST http://192.168.0.165:1400/DeviceProperties/Control
2016-04-11 17:02:00.233 [DEBUG] [e.s.i.t.upnp.UpnpIOServiceImpl:276  ] - Current state of service prevents invoking that action. Connection error or no response received.
2016-04-11 17:02:00.234 [DEBUG] [.b.s.handler.ZonePlayerHandler:571  ] - Sonos player  is not available in local network
2016-04-11 17:02:00.234 [INFO ] [ome.event.ThingStatusInfoEvent:43   ] - 'sonos:CONNECT:BedroomLeontine' updated: OFFLINE